### ISSUE TITLE
Duplicated content issue on non-url-rewritten category pages

### ISSUE THREAD

**Issue by hostep** [clerkio/clerk-magento2#81]
Hi

We have this very weird issue, it requires visiting a category url that has no url rewrite
When we do this, we can see that the ajax call to `clerk/powerstep/popup` gives us back the entire html dom of the category page and inserts it again underneath the already existing page. Which causes the page's DOM being duplicated entirely and we have two times the header, menu, products, footer, ...

Seen with:
- Magento OS 2.3.6-p1
- Clerk module 4.6.1 or 4.6.2

Steps to reproduce:
- Enable the Powerstep feature with type Popup
- Have a category with a bunch of products
- Visit the category url which looks like this: `https://domain.example/{storeview_code}/catalog/category/view/s/{category-url-key}/id/{category-entity-id}/` (you can probably quickly simulate this by removing url rewrites for this category from the `url_rewrite` database table and flushing your cache and then clicking on that category in the menu
- Seeing that the entire html is being duplicated after the ajax call to the powerstep popup finished

This is not happening when a category has a "nice" url from the url rewrites.
We are currently investigating on this shop why certain categories do not have such an url, and are going to fix this, but this is a weird side effect we noticed and thought it might be a good idea to at least document it publicly in case somebody else runs into this as well.
Not sure if it makes sense to try to fix this, I'll leave that up to you to decide.

Thanks!

**Comment by FoulBachelor** [comment]
Hi Hotep,

This is an interesting case you have found here. Does the issue persist if you rename the block for the categories in question?

I ask, because I have previously seen some page content duplication issues which would occur due to an invalid require[] js dependency, rather than a block name issue.

**Comment by FoulBachelor** [comment]
Thanks for the additional info, I have managed to reproduce it here:

https://mage2.stubbe.dev/catalog/category/view/s/gear/id/4/

Clearing the DB entry was a good suggestion.

It seems like the issue occurs from the call to getPowerstepPopup() in powerstep_script.phtml which is only gated by page ready, receives and empty page from the request to the block template, it it gets the redirect for its current page instead and appends it.

This second init in the script file is only there to account for users who disable ajaxAddToCart, and simply have a page reload or cart redirect.

In order to avoid the issue for now, I have gated it by looking for the success message in the DOM.

https://github.com/clerkio/clerk-magento2/commit/49a96d63b888e3809f28e14759605b924bb51394
